storage-provider/server/src/db.rs

```diff
@@ -252,6 +285,28 @@ impl DealDB {
         Ok(Some(unsealed_sector))
     }
 
+    /// Removes and returns a given [`UnsealedSector`], if it doesn't exist, returns `None`.
+    /// Locks the passed key for the duration of this operation!
+    ///
+    /// Removal is only done on success!
+    pub fn remove_unsealed_sector(
+        &self,
+        sector_number: SectorNumber,
+    ) -> Result<Option<UnsealedSector>, DBError> {
+        self.remove_sector(sector_number, UNSEALED_SECTORS_CF)
+    }
```

Comment:
Considering this is only used to "move" (remove and insert) the sector between column families, I wonder if we should make the public API encapsulate both remove and insert for a slightly more safer usage